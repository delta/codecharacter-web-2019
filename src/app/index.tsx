import Dashboard from 'app/containers/Dashboard';
// @ts-ignore
import { initializeRendererAssets } from 'codecharacter-renderer';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

initializeRendererAssets();

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
