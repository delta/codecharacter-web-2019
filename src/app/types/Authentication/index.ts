export enum AuthType {
  LOGIN = 'LOGIN',
  REGISTER = 'REGISTER',
}

export interface OwnProps {
  setIsAuthenticationOpen: (isAuthenticationOpen: boolean) => void;
}

export interface State {
  authType: AuthType;
}

export interface StateProps {
  isLoggedIn: boolean;
}

export interface DispatchProps {
  updateErrorMessage: (errorMessage: string) => void;
  toggleUserProfileModal: (isUserProfileModalOpen: boolean) => void;
}

export type Props = OwnProps & StateProps & DispatchProps;
