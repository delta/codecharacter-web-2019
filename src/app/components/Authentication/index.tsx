import { faTimes, faUser, faUserPlus } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Login from 'app/containers/Authentication/Login';
import Register from 'app/containers/Authentication/Register';
import * as styles from 'app/styles/Authentication.module.css';
import * as AuthenticationInterfaces from 'app/types/Authentication';
import classnames from 'classnames';
import * as React from 'react';
import { Button, Col, Grid, Row } from 'react-bootstrap';
export class Authentication extends React.Component<
  AuthenticationInterfaces.Props,
  AuthenticationInterfaces.State
> {
  constructor(props: AuthenticationInterfaces.Props) {
    super(props);
    this.state = {
      authType: AuthenticationInterfaces.AuthType.LOGIN,
    };
  }

  public render() {
    const { authType } = this.state;
    const { isLoggedIn, updateErrorMessage, setIsAuthenticationOpen } = this.props;
    return (
      <div
        className={classnames(styles.Authentication, {
          [`${styles.modalOpen}`]: !isLoggedIn,
        })}
      >
        <Grid
          fluid={true}
          style={{
            width: authType === AuthenticationInterfaces.AuthType.LOGIN ? '500px' : '850px',
          }}
          className={classnames(styles.modal, {
            [`${styles.slideEnter}`]: !isLoggedIn,
          })}
        >
          <Row className="mt-2">
            <Col sm={11} />
            <Col
              className={'text-right'}
              sm={1}
              style={{ color: 'black', cursor: 'pointer' }}
              onClick={() => {
                setIsAuthenticationOpen(false);
              }}
            >
              <FontAwesomeIcon icon={faTimes} />
            </Col>
          </Row>
          <Row className="mt-1">
            <Col sm={12}>
              <div className={classnames(styles.buttonPanelBackground)}>
                <Button
                  className={classnames(styles.buttonPanel, {
                    [`${styles.buttonPanelActive}`]:
                      authType === AuthenticationInterfaces.AuthType.LOGIN,
                  })}
                  onClick={() => {
                    updateErrorMessage('');
                    this.handleSelectPanel(AuthenticationInterfaces.AuthType.LOGIN);
                  }}
                >
                  <FontAwesomeIcon icon={faUser} /> &nbsp;Login
                </Button>
                <Button
                  className={classnames(styles.buttonPanel, {
                    [`${styles.buttonPanelActive}`]:
                      authType === AuthenticationInterfaces.AuthType.REGISTER,
                  })}
                  onClick={() => {
                    updateErrorMessage('');
                    this.handleSelectPanel(AuthenticationInterfaces.AuthType.REGISTER);
                  }}
                >
                  <FontAwesomeIcon icon={faUserPlus} /> &nbsp;Register
                </Button>
              </div>
            </Col>
          </Row>
          {authType === AuthenticationInterfaces.AuthType.LOGIN ? (
            <Login handleSelectPanel={this.handleSelectPanel} />
          ) : (
            <Register handleSelectPanel={this.handleSelectPanel} />
          )}
        </Grid>
      </div>
    );
  }

  private handleSelectPanel = (authType: AuthenticationInterfaces.AuthType) => {
    this.setState({
      authType,
    });
  };
}
