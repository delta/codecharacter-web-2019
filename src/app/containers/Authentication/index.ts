import { UserActions } from 'app/actions';
import { Authentication } from 'app/components/Authentication';
import { RootState } from 'app/reducers';
import * as AuthenticationInterfaces from 'app/types/Authentication/';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    isLoggedIn: rootState.user.isLoggedIn,
  };
};

const authenticationContainer = connect<
  AuthenticationInterfaces.StateProps,
  AuthenticationInterfaces.DispatchProps,
  {}
>(mapStateToProps, {
  toggleUserProfileModal: UserActions.toggleUserProfileModal,
  updateErrorMessage: UserActions.updateErrorMessage,
})(Authentication);

export default authenticationContainer;
