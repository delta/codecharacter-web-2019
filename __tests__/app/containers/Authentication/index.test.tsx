import Authentication from 'app/containers/Authentication';
import { configureStore } from 'app/store';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('Authentication main Container', () => {
  const { store } = configureStore();
  const wrapper = shallow(<Authentication setIsAuthenticationOpen={(v: boolean) => undefined} />, {
    context: {
      store,
    },
  });

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
