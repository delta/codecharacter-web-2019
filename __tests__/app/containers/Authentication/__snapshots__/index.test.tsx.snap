// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication main Container Should render 1`] = `
<Authentication
  isLoggedIn={false}
  setIsAuthenticationOpen={[Function]}
  toggleUserProfileModal={[Function]}
  updateErrorMessage={[Function]}
/>
`;
